nx/lib/nx.ex

```diff
@@ -7387,6 +7409,12 @@ defmodule Nx do
     indexed_op(target, indices, updates, :indexed_put)
   end
 
+  defp indexed_op(target, %Nx.Tensor{shape: {_}} = index, update, op) when is_tensor(update) do
```

Comment:
@josevalim in theory we could extend this to accept a list of numbers/tensors as well in the `index`. WDYT?

Would be something like `index = if is_list(index), do: Enum.map(index, &to_tensor/1) |> Nx.concatenate(), else: to_tensor(index)`